.gitignore

```diff
@@ -3,10 +3,15 @@ build
 debug
 __pycache__
 catalyst.egg-info
+*mlir
+*\.ll
+*\.o
+*\.so
```

Comment:
I'm not a fan of these blanket ignores. For one we actually have `.mlir` files in our repo, but in general if we generate additional build files with the code in this PR those should go in a specific directory which we can then exclude.